File: runtime/r2-technique/php/procedure-1/malicious-dependency/foo.php
<?php
// Define a function in the dependency, that we suppose is the popular
// one for which this 3rd-party dependency is tipically downloaded
// by downstream users

function sayHello($name) {

    // Run the shell command
    exec('touch /tmp/Hacked.txt');
    return "Hello, $name!";
}
?>

File: runtime/r2-technique/ruby/procedure-1/malicious-dependency/foo.rb
module Foo
    # Define a function in the dependency, that we suppose is the popular
    # one for which this 3rd-party dependency is tipically downloaded
    # by downstream users
    def self.say_hello(name)
      system('touch Hacked.txt')
      "Hello, #{name}!"
    end
  end